### PR TITLE
docs:handle the console on Windows

### LINKED ISSUE #3235: document how to prevent a console to open on Windows
We need to add documentation for windows_subsystem attribute in our rust documentation.
See #1190
And also how to keep log messages when wanted.

### Original:

How to keep output message in console, but not to open a console window when run the app? 

I disable the console window with the solution in  #1190 .
However, I use clap to let my app got some cli parameters and print help message.  The code 
`    #![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]`
    totally disable all print info.
Is there any solution to keep the print info when I'm in a console, but not to open a new one if not?

### PR DESCRIPTION
**Pull request by asuper0**
Closes #3235. I left the Node.js example with TODO while I am not familiar with it.

### CODE REVIEW

File: docs/src/content/docs/platforms/desktop.mdx [hunk 1]
```diff
@@ -15,6 +17,85 @@ are supported by their vendors at the time of a Slint version release.
...
+1. Open the project's Property Pages dialog box. For details, see Set C++ compiler and build properties in Visual Studio.
+
+2. Select the Configuration Properties > Linker > System property page.
+
+3. Modify the SubSystem property, select "Window(/SUBSYSTEM:WINDOWS)".
+
+</TabItem>
+<TabItem label="Node.js">
+
+TODO: Add node.js
```
Comment (NigelBreslaw) [inline docs/src/content/docs/platforms/desktop.mdx:54]: As we are close to publishing these docs can we get the NodeJS info or just not show the tab for now? There should not be todo's in the docs for a published release.

File: docs/src/content/docs/platforms/desktop.mdx [hunk 2]
```diff
@@ -15,6 +17,85 @@ are supported by their vendors at the time of a Slint version release.
...
+
+    if run_without_console {
+        unsafe {
+            winapi::um::wincon::FreeConsole();
+        }
+    }
+}
+```
+
+See more details at [#3235](https://github.com/slint-ui/slint/issues/3235)
```
Comment (NigelBreslaw) [inline docs/src/content/docs/platforms/desktop.mdx:97]: I'm trying to limit the issue links we have in the docs to just active issues/bugs. If the info is relevant it should be here in the docs?

File: docs/src/content/docs/platforms/desktop.mdx (hunk 1 above)
Comment (asuper0) [reply docs/src/content/docs/platforms/desktop.mdx:54]: Yes, indeed. But I am not familiar with it. Hope some one else could help. 🤣 I believe this task is easy for skilled Node.js developers.

File: docs/src/content/docs/platforms/desktop.mdx [hunk 3]
```diff
@@ -15,6 +17,85 @@ are supported by their vendors at the time of a Slint version release.
...
+3. Modify the SubSystem property, select "Window(/SUBSYSTEM:WINDOWS)".
+
+</TabItem>
+<TabItem label="Node.js">
+
+TODO: Add node.js
+</TabItem>
+</Tabs>
+
+Set subsystem to "windows" would lost stdout if you run the application from command line. If you want to deal with CLI, consider use `FreeConsole()` instead of flag subsystem.
```
Comment (NigelBreslaw) [inline docs/src/content/docs/platforms/desktop.mdx:58]: This is a great PR so forgive some little tweaks to just make the language a bit more readable. Is this trying to say "When running the application from the command line, if the subsystem is set to `windows` it will no longer output stdout. To get it back consider using `FreeConsole()'." 
It's a bit hard to parse the current sentence.

File: docs/src/content/docs/platforms/desktop.mdx (hunk 2 above)
Comment (asuper0) [reply docs/src/content/docs/platforms/desktop.mdx:97]: The reason I left this link is [the discussion](https://github.com/slint-ui/slint/issues/3235#issuecomment-1680032269). I want the reader to understand why I didn't adopt `AttachConsole()` solution. But listing this discussion in full would make the document a bit lengthy.

File: docs/src/content/docs/platforms/desktop.mdx [hunk 4]
```diff
@@ -15,6 +17,85 @@ are supported by their vendors at the time of a Slint version release.
...
+
+Here is a Rust example. The parent process should be "explorer.exe" if you run the application from desktop, or folder window, so we could free the console as we dont want to see it.
```
Comment (NigelBreslaw) [inline docs/src/content/docs/platforms/desktop.mdx:60]: Again this is hard to parse. "The following Rust example.. <does?>". I'm the UI person here so my knowledge of what is being solved here is limited! Is it something like "The following Rust example detects where the app is being run from. If run from the command line `FreeConsole()` is used to let it output debug info while not outputting anything when run as a traditional app."

File: docs/src/content/docs/platforms/desktop.mdx (hunk 2 above)
Comment (NigelBreslaw) [reply docs/src/content/docs/platforms/desktop.mdx:97]: Good point. Let's leave it as it is then.

File: docs/src/content/docs/platforms/desktop.mdx (hunk 1 above)
Comment (NigelBreslaw) [reply docs/src/content/docs/platforms/desktop.mdx:54]: I will raise an issue for myself to source this and make sure it's not left in the doc 😁

File: docs/src/content/docs/platforms/desktop.mdx (hunk 3 above)
Comment (asuper0) [reply docs/src/content/docs/platforms/desktop.mdx:58]: Yes, that's what I mean. Sorry, English is not my mother tongue.

File: docs/src/content/docs/platforms/desktop.mdx (hunk 4 above)
Comment (asuper0) [reply docs/src/content/docs/platforms/desktop.mdx:60]: Partially right. If run from the command line, we don't use `FreeConsole()` because we need it. Only use `FreeConsole()` when run as a traditional app, so the console window will be closed.

Comment (NigelBreslaw) [reply docs/src/content/docs/platforms/desktop.mdx:60]: "Here is a Rust example. It checks if the parent process is "explorer.exe" to ensure the console window is closed when opening the app via the normal windows GUI". Am I understanding now?

Comment (NigelBreslaw) [reply docs/src/content/docs/platforms/desktop.mdx:60]: So you want to make these tweaks? Then we get this approved and merged in.

Comment (asuper0) [reply docs/src/content/docs/platforms/desktop.mdx]: Sure, I used your improvement and submitted it

File: docs/src/content/docs/platforms/desktop.mdx [hunk 5]
```diff
@@ -15,6 +17,85 @@ are supported by their vendors at the time of a Slint version release.
 | Windows 10       | x86-64       |
 | Windows 11       | x86-64       |
 
+### Handle the console window
+
+When you run an application built by slint, a console window would show, by default ( see [/SUBSYSTEM (Specify Subsystem)](https://learn.microsoft.com/en-us/cpp/build/reference/subsystem-specify-subsystem?view=msvc-170)).
+
+You could disable the console by specify window subsystem.
```
Comment (tronical) [inline docs/src/content/docs/platforms/desktop.mdx:24]:
```suggestion
You could disable the console by specify `WINDOWS` subsystem.
```

File: docs/src/content/docs/platforms/desktop.mdx [hunk 6]
```diff
@@ -15,6 +17,85 @@ are supported by their vendors at the time of a Slint version release.
...
+
+<Tabs syncKey="dev-language">
+<TabItem label="Rust">
+
+Add the code to the top of .rs file which contains `fn main()`.
```
Comment (tronical) [inline docs/src/content/docs/platforms/desktop.mdx:29]:
```suggestion
Add the code to the top of .rs file which contains `fn main()`:
```

File: docs/src/content/docs/platforms/desktop.mdx [hunk 7]
```diff
@@ -15,6 +17,85 @@ are supported by their vendors at the time of a Slint version release.
...
+<TabItem label="C++">
+
+Add link flag `/SUBSYSTEM:WINDOWS`. If you are using Visual Studio, follow the steps:
+
+1. Open the project's Property Pages dialog box. For details, see Set C++ compiler and build properties in Visual Studio.
+
+2. Select the Configuration Properties > Linker > System property page.
+
+3. Modify the SubSystem property, select "Window(/SUBSYSTEM:WINDOWS)".
+
```
Comment (tronical) [inline docs/src/content/docs/platforms/desktop.mdx:50]: We typically recommend the use of CMake with C++. So I suggest the following:

```````suggestion

Select the `WINDOWS` subsystem by setting the [`WIN32_EXECUTABLE`](https://cmake.org/cmake/help/latest/prop_tgt/WIN32_EXECUTABLE.html#prop_tgt:WIN32_EXECUTABLE) target property on your executable target:

```cmake
add_executable(my_program ...)
set_property(TARGET my_program APPEND PROPERTY WIN32_EXECUTABLE TRUE)
```

```````

**Review by NigelBreslaw — APPROVED** [review]

File: docs/src/content/docs/platforms/desktop.mdx [hunk 8]
```diff
@@ -15,6 +17,85 @@ are supported by their vendors at the time of a Slint version release.
...
+        _ => false,
+    };
+
+    if run_without_console {
+        unsafe {
+            winapi::um::wincon::FreeConsole();
+        }
+    }
+}
+```
```
[the remark this answers is not in the document]
Comment (NigelBreslaw) [reply docs/src/content/docs/platforms/desktop.mdx:95]: I'm going to merge this in as is. @asuper0 we will be tweaking the docs quite a bit to get them in shape for 1.9 so I can remove or move the extra rust info then. Main thing is to have all the new content. Thanks again for getting this added.

[the remark this answers is not in the document]
Comment (tronical) [reply docs/src/content/docs/platforms/desktop.mdx:95]: 👍👍 please merge and thanks @asuper0 !
